UserDetails/forms.py

```diff
@@ -29,8 +29,10 @@ class Meta:
         fields = ('username', 'password1', 'password2', 'email')
 
     def clean(self):
+        cleaned_data = super(RegisterUserForm, self).clean()
+
         # Check if the email is not already used.
-        email = self.cleaned_data.get('email')
+        email = cleaned_data.get('email')
         if User.objects.filter(email=email).exists():
             msg = 'E-mail is already used.'
             self._errors['email'] = self.error_class([msg])
```

Comment:
I know this is not in the changeset, but I would use `raise ValidationError` or `add_error` here. It's not really good practice to modify private variables.